importer.go

```diff
@@ -155,7 +155,7 @@ func (i *Importer) open(imp string) (io.ReadCloser, error) {
 		return r, err
 	}
 
-	resp, err := i.fetcher().Get("https:" + imp)
+	resp, err := i.fetcher().Get("https:" + imp) //nolint: bodyclose
```

Comment:
I thought this was a real bug - not closing the body. Now I look at it, I see this function returns the body and the caller closes it. Can we make this comment:
```
//nolint: bodyclose // caller closes body
```
?

... and now I look at it closer, I see there is a bug. The are a couple of code paths where `nil` is returned and the body is not closed. This needs @samuong's cancelable closer perhaps: https://github.com/samuong/alpaca/tree/master/cancelable (and that closer needs some godocs, Sam). That closer could go into snippets. Maybe `closeIgnoreErr` should too and it should be in some sort of "closer" package?

Reply:
Too much for this PR. Leaving for later.

Reply:
I'm assuming `open()` is only called from `fetch()`. Instead of using a `cancelable.Closer`, why not change the `open()` function to return a `[]byte`, so that callers don't have to worry about closing it?

1. Use `ioutil.ReadFile()` for the file case, which handles closing the file for you.
2. For the net path case, move the `ioutil.ReadAll()` call into `open()`, and defer the close like you normally would.
3. Rename it from `open` to something else, like `read`,
4. Consider moving what's left of `fetch()` into `readViaCache()`,

Reply:
I have implemented this and saved two lines! (+22, -24). But I have also split `open`/`read` into non-method functions at the leaves so I'll update some tests too. I also noticed I was not testing the caching of `noContent` so I am adding a test for that too.

This will be on a new PR because I don't know how to unmerge this one without upsetting the git gods.

Thanks Sam.